pkg/controllers/dnspolicy/dnspolicy_dnsrecords.go

```diff
@@ -25,15 +25,15 @@ func (r *DNSPolicyReconciler) reconcileDNSRecords(ctx context.Context, dnsPolicy
 
 	for _, gw := range gwDiffObj.GatewaysWithInvalidPolicyRef {
 		log.V(1).Info("reconcileDNSRecords: gateway with invalid policy ref", "key", gw.Key())
-		err := r.deleteGatewayDNSRecords(ctx, gw.Gateway, dnsPolicy)
+		err := r.deleteDNSRecords(ctx, dnsPolicy)
```

Comment:
This will need to use both the gateway and policy labels since it should only be deleting resources created for the gateway being processed (gw.Gateway). What `deleteGatewayDNSRecords` was doing originally was correct in this scenario.